File: Leetcode/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # for i in range(len(nums)):
        #     for j in range(i+1,len(nums)):
        #         if(nums[i]+nums[j]==target):
        #             return [i,j]
        # return [-1,-1]
        mp={}
        result=[]
        for i in range(len(nums)):
            mp[nums[i]]=i

        for i in range(len(nums)):
            if target-nums[i] in mp and mp[target-nums[i]]!=i:
                return [i,mp[target-nums[i]]]

        return [-1,-1] if result==[] else result

        

File: Leetcode/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int n=nums.length;
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<n;i++){
            map.put(nums[i],i);
        }

        for(int i=0;i<n;i++){
            
            if(map.containsKey(target-nums[i])){
                if(map.get(target-nums[i])==i) continue;
                return new int[]{i,map.get(target-nums[i])};
            }
        }

        return new int[]{-1,-1};


    }
}